### PR TITLE
Sam/fiat-plugin-cleanup

### PR DESCRIPTION
**Pull request by samholmes**
### CHANGELOG

 none

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204378499731823

### CODE REVIEW

File: src/plugins/gui/fiatPluginTypes.ts [hunk 1]
```diff
@@ -101,6 +102,7 @@ export interface FiatPluginUi {
   addressForm: (params: FiatPluginAddressFormParams) => Promise<HomeAddress>
   sepaForm: (params: FiatPluginSepaFormParams) => Promise<SepaInfo>
   sepaTransferInfo: (params: FiatPluginSepaTransferParams) => Promise<void>
+  navigation: NavigationBase
```
Comment (swansontec) [inline src/plugins/gui/fiatPluginTypes.ts:105]: This ties the plugins pretty deeply to our app's internals. Is there some other way?

File: src/plugins/gui/fiatPluginTypes.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export interface FiatPluginEnterAmountParams {
   label1: string
   label2: string
   convertValue: (sourceFieldNum: number, value: string) => Promise<string | undefined>
-  onSubmit: (value: FiatPluginEnterAmountResponse) => Promise<void>
+  onSubmit: (value: FiatPluginEnterAmountResponse) => void
```
Comment (swansontec) [inline src/plugins/gui/fiatPluginTypes.ts:38]: We generally want things like `onSubmit` to accept a promise, and then show a spinner as long as the promise is pending. This provides feedback to the user with essentially zero developer effort. It also comes with the expectation that the scene or component will catch any errors and do `showError` on the callback's behalf. Otherwise you have to remember to do that manually, which humans are pretty bad at.

File: src/components/common/BrandLogo.tsx
Comment (swansontec) [inline src/components/common/BrandLogo.tsx:1]: This file doesn't really belong in this PR, as you mentioned.

**Review by swansontec — CHANGES_REQUESTED** [review]
These are more architectural questions that coding mistakes.

**Comment by samholmes** [comment]
Rebase diff:

```diff
diff --git a/src/components/common/BrandLogo.tsx b/src/components/common/BrandLogo.tsx
deleted file mode 100644
index f2a74676a..000000000
--- a/src/components/common/BrandLogo.tsx
+++ /dev/null
@@ -1,16 +0,0 @@
-import React from 'react'
-import { Image } from 'react-native'
-
-import { styled } from '../hoc/styled'
-import { useTheme } from '../services/ThemeContext'
-
-export const BrandLogo = () => {
-  const theme = useTheme()
-
-  return <Logo source={theme.primaryLogo} resizeMode="contain" />
-}
-
-const Logo = styled(Image)(props => ({
-  display: 'flex',
-  alignSelf: 'center'
-}))
diff --git a/src/plugins/gui/amountQuotePlugin.ts b/src/plugins/gui/amountQuotePlugin.ts
index b23272f8c..34c312f80 100644
--- a/src/plugins/gui/amountQuotePlugin.ts
+++ b/src/plugins/gui/amountQuotePlugin.ts
@@ -12,14 +12,7 @@ import { getTokenId } from '../../util/CurrencyInfoHelpers'
 import { fetchInfo } from '../../util/network'
 import { logEvent } from '../../util/tracking'
 import { fuzzyTimeout } from '../../util/utils'
-import {
-  FiatPlugin,
-  FiatPluginEnterAmountResponse,
-  FiatPluginFactory,
-  FiatPluginFactoryArgs,
-  FiatPluginGetMethodsResponse,
-  FiatPluginStartParams
-} from './fiatPluginTypes'
+import { FiatPlugin, FiatPluginFactory, FiatPluginFactoryArgs, FiatPluginGetMethodsResponse, FiatPluginStartParams } from './fiatPluginTypes'
 import { FiatProvider, FiatProviderAssetMap, FiatProviderGetQuoteParams, FiatProviderQuote } from './fiatProviderTypes'
 import { createStore, getBestError, getRateFromQuote } from './pluginUtils'
 import { banxaProvider } from './providers/banxaProvider'
@@ -152,7 +145,7 @@ export const amountQuoteFiatPlugin: FiatPluginFactory = async (params: FiatPlugi
 
       let enterAmountMethods: FiatPluginGetMethodsResponse
       // Navigate to scene to have user enter amount
-      await showUi.enterAmount({
+      showUi.enterAmount({
         headerTitle: isBuy ? sprintf(lstrings.fiat_plugin_buy_currencycode, currencyCode) : sprintf(lstrings.fiat_plugin_sell_currencycode_s, currencyCode),
         isBuy,
         label1: sprintf(lstrings.fiat_plugin_amount_currencycode, displayFiatCurrencyCode),
@@ -290,11 +283,11 @@ export const amountQuoteFiatPlugin: FiatPluginFactory = async (params: FiatPlugi
             return toFixed(bestQuote.fiatAmount, 0, 2)
           }
         },
-        onSubmit: (value: FiatPluginEnterAmountResponse) => {
+        async onSubmit() {
           if (bestQuote == null) {
             return
           }
-          bestQuote.approveQuote({ showUi, coreWallet }).catch(showUi.showError)
+          await bestQuote.approveQuote({ showUi, coreWallet })
         }
       })
     }
diff --git a/src/plugins/gui/fiatPlugin.tsx b/src/plugins/gui/fiatPlugin.tsx
index 446050464..af39ba5b4 100644
--- a/src/plugins/gui/fiatPlugin.tsx
+++ b/src/plugins/gui/fiatPlugin.tsx
@@ -62,7 +62,7 @@ export const executePlugin = async (params: {
       ))
       return result
     },
-    enterAmount: (params: FiatPluginEnterAmountParams) => {
+    enterAmount(params: FiatPluginEnterAmountParams) {
       const { headerTitle, label1, label2, initialAmount1, convertValue, getMethods, onSubmit } = params
       logEvent(isBuy ? 'Buy_Quote' : 'Sell_Quote')
 
@@ -132,7 +132,6 @@ export const executePlugin = async (params: {
         })
       })
     },
-    navigation,
     popScene: async () => {
       navigation.pop()
     }
diff --git a/src/plugins/gui/fiatPluginTypes.ts b/src/plugins/gui/fiatPluginTypes.ts
index cdf0845dd..3f875e3e8 100644
--- a/src/plugins/gui/fiatPluginTypes.ts
+++ b/src/plugins/gui/fiatPluginTypes.ts
@@ -5,7 +5,6 @@ import { DisablePluginMap } from '../../actions/ExchangeInfoActions'
 import { SendScene2Params } from '../../components/scenes/SendScene2'
 import { HomeAddress, SepaInfo } from '../../types/FormTypes'
 import { GuiPlugin } from '../../types/GuiPluginTypes'
-import { NavigationBase } from '../../types/routerTypes'
 import { EdgeTokenId } from '../../types/types'
 import { EnterAmountPoweredBy } from './scenes/FiatPluginEnterAmountScene'
 
@@ -35,7 +34,7 @@ export interface FiatPluginEnterAmountParams {
   label1: string
   label2: string
   convertValue: (sourceFieldNum: number, value: string) => Promise<string | undefined>
-  onSubmit: (value: FiatPluginEnterAmountResponse) => void
+  onSubmit: (value: FiatPluginEnterAmountResponse) => Promise<void>
   getMethods?: (methods: FiatPluginGetMethodsResponse) => void
   initialAmount1?: string
   headerIconUri?: string
@@ -102,7 +101,6 @@ export interface FiatPluginUi {
   addressForm: (params: FiatPluginAddressFormParams) => Promise<HomeAddress>
   sepaForm: (params: FiatPluginSepaFormParams) => Promise<SepaInfo>
   sepaTransferInfo: (params: FiatPluginSepaTransferParams) => Promise<void>
-  navigation: NavigationBase
   popScene: () => {}
   send: (params: SendScene2Params) => Promise<void>
   // showWebView: (params: { webviewUrl: string }) => Promise<void>
diff --git a/src/plugins/gui/scenes/FiatPluginEnterAmountScene.tsx b/src/plugins/gui/scenes/FiatPluginEnterAmountScene.tsx
index ac901b412..34d443663 100644
--- a/src/plugins/gui/scenes/FiatPluginEnterAmountScene.tsx
+++ b/src/plugins/gui/scenes/FiatPluginEnterAmountScene.tsx
@@ -5,6 +5,7 @@ import IonIcon from 'react-native-vector-icons/Ionicons'
 
 import { Card } from '../../../components/cards/Card'
 import { SceneWrapper } from '../../../components/common/SceneWrapper'
+import { showError } from '../../../components/services/AirshipInstance'
 import { cacheStyles, Theme, useTheme } from '../../../components/services/ThemeContext'
 import { MainButton } from '../../../components/themed/MainButton'
 import { OutlinedTextInput } from '../../../components/themed/OutlinedTextInput'
@@ -17,7 +18,7 @@ import { FiatPluginEnterAmountResponse, FiatPluginGetMethodsResponse } from '../
 
 export interface FiatPluginEnterAmountParams {
   headerTitle: string
-  onSubmit: (response: FiatPluginEnterAmountResponse) => void
+  onSubmit: (response: FiatPluginEnterAmountResponse) => Promise<void>
   label1: string
   label2: string
   onChangeText: (fieldNum: number, value: string) => Promise<void>
@@ -110,7 +111,7 @@ export const FiatPluginEnterAmountScene = React.memo((props: Props) => {
     })
   })
   const handleSubmit = useHandler(() => {
-    onSubmit({ lastUsed: lastUsed.current, value1, value2 })
+    onSubmit({ lastUsed: lastUsed.current, value1, value2 }).catch(showError)
   })
 
   let statusTextStyle = styles.text

```

**Review by swansontec — APPROVED** [review]
